### PR TITLE
feat: state data for aborted tx amount

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

- Adds new state data to track the total amount of zeta tokens that have been aborted 
- This value is accounted for the zeta supply checker

Closes: <PD-XXXX>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: app/setup_handlers.go [hunk 1]
```diff
@@ -17,7 +18,7 @@ func SetupHandlers(app *App) {
 		for m, mb := range app.mm.Modules {
 			vm[m] = mb.ConsensusVersion()
 		}
-		vm[observertypes.ModuleName] = vm[observertypes.ModuleName] - 1
+		vm[observertypes.ModuleName] = vm[crosschaintypes.ModuleName] - 1
```
Comment (lumtis) [inline app/setup_handlers.go:21]:
```suggestion
		vm[crosschaintypes.ModuleName] = vm[crosschaintypes.ModuleName] - 1
```

File: proto/crosschain/cross_chain_tx.proto [hunk 2]
```diff
@@ -32,6 +32,13 @@ message InboundTxParams {
   uint64 inbound_tx_finalized_zeta_height = 10;
 }
 
+message AbortedZetaAmount {
+  string amount = 6 [
```
Comment (lumtis) [inline proto/crosschain/cross_chain_tx.proto:36]:
```suggestion
  string amount = 1 [
```

File: proto/crosschain/query.proto [hunk 3]
```diff
@@ -147,6 +147,19 @@ service Query {
...
+
+  rpc AbortedZetaAmount(QueryAbortedZetaAmountRequest) returns (QueryAbortedZetaAmountResponse) {
+    option (google.api.http).get = "/zeta-chain/crosschain/abortedZetaAmount";
+  }
+}
+
+message QueryAbortedZetaAmountRequest {}
+
+message QueryAbortedZetaAmountResponse {
+  string amount = 6 [
```
Comment (lumtis) [inline proto/crosschain/query.proto:159]:
```suggestion
  string amount = 1 [
```

File: zetaclient/zeta_supply_checker.go [hunk 4]
```diff
@@ -174,18 +174,17 @@ func ValidateZetaSupply(logger zerolog.Logger, abortedTxAmounts, zetaInTransit,
 	return true
 }
 
+// TODO : Add genesis state for Aborted amount in zeta
+// TODO : Add tests for keeper functions
+// TODO : Add cli commands for querying aborted amount
```
Comment (lumtis) [inline zetaclient/zeta_supply_checker.go:179]: These TODOs are solved in this PR isn't it?

**Review by lumtis — COMMENTED** [review]
It looks good to me.
I'm just wondering, for future-proofing purposes, maybe we could use a more general state variable `ZetaAccounting` instead `AbortedZetaAmount` and add `abortedAmount` in this state variable. It would reduce complexity and refactoring work in case we want to add a new variable for accounting purpose in the store in the future.
What do you think?

File: docs/cli/zetacored/zetacored_query_crosschain_get-aborted-zeta-amount.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# query crosschain get-aborted-zeta-amount
+
+Query current tss address
```
Comment (brewmaster012) [inline docs/cli/zetacored/zetacored_query_crosschain_get-aborted-zeta-amount.md]: Description is wrong

File: x/crosschain/keeper/keeper_cross_chain_tx.go [hunk 6]
```diff
@@ -51,6 +47,9 @@ func (k Keeper) SetCctxAndNonceToCctxAndInTxHashToCctx(ctx sdk.Context, send typ
 			Tss:       tss.TssPubkey,
 		})
 	}
+	if send.CctxStatus.Status == types.CctxStatus_Aborted {
+		k.AddZetaAccounting(ctx, send.GetCurrentOutTxParam().Amount)
```
Comment (brewmaster012) [inline x/crosschain/keeper/keeper_cross_chain_tx.go:51]: 1. The function name AddZetaAccounting can be more specific. 
2. Do you need to test whether the outbound is supposed to be ZETA token transfer, not some other tokens?

**Review by brewmaster012 — CHANGES_REQUESTED** [review]

**Review by kevinssgh — APPROVED** [review]
Just a typo in a file name: x/crosschain/keeper/zeta_accountin.go

File: x/crosschain/keeper/keeper_cross_chain_tx.go (hunk 6 above)
Comment (kingpinXD) [reply x/crosschain/keeper/keeper_cross_chain_tx.go]: Thanks , good catch !

**Pull request closed without merging**